ingestion/src/legacy/raw.rs

```diff
@@ -0,0 +1,421 @@
+// of empty entries that are probably obsinn trying to tell us to delete things...
+const QUERY_STR: &str = r#"
+    INSERT INTO legacy.data
+        (timeseries, obstime, original)
+    VALUES($1, $2, $3)
+    ON CONFLICT ON CONSTRAINT data_pkey
+        DO NOTHING
+"#;
+
+const NONSCALAR_QUERY_STR: &str = r#"
+    INSERT INTO public.nonscalar_data
+        (timeseries, obstime, obsvalue)
+    VALUES($1, $2, $3)
+    ON CONFLICT ON CONSTRAINT nonscalar_data_pkey
+        DO NOTHING
+"#;
+
+#[derive(Error, Debug)]
+pub enum Error {
+    #[error("kafka returned an error: {0}")]
+    Kafka(#[from] KafkaError),
+    #[error("failed to determine format of kafka message")]
+    Format,
+    #[error("failed to decode kafka message as Utf8")]
+    Utf8(#[from] std::str::Utf8Error),
+    #[error("failed to parse kldata message: {0}")]
+    Parse(#[from] ParseError),
+    #[error("postgres returned an error: {0}")]
+    Database(#[from] tokio_postgres::Error),
+    #[error(transparent)]
+    Common(#[from] common::Error),
+}
+
+type Datum = CommonDatum<ObsType>;
+type UnlabelledDatum = CommonUnlabelledDatum<ObsType>;
+
+// we have to do this with the u8 slice because some messages on the topic
+// (bufr) cannot be decoded as utf8
+fn is_kldata_message(message: &[u8]) -> Result<bool, Error> {
+    let format_raw = message
+        .split(|elem| {
+            let char = *elem as char;
+            char == '\n' || char == '/'
+        })
+        .next()
+        .ok_or(Error::Format)?;
+
+    let format = std::str::from_utf8(format_raw)?;
+
+    Ok(format.trim() == "kldata")
+}
+
+// modified version of kldata::parse_obs that returns RawDatum instead of ObsinnChunk
+fn parse_obs(
+    csv_body: Lines,
+    columns: &[ObsinnId],
+    reference_params: ParamConversions,
+    header: ObsinnHeader,
+) -> Result<Vec<UnlabelledDatum>, ParseError> {
+    let mut obs = Vec::new();
+
+    for row in csv_body {
+        let (timestamp, vals) = {
+            let mut vals = row.split(',');
+
+            let raw_timestamp = vals.next().ok_or(ParseError::EmptyRow)?;
+
+            // TODO: timestamp parsing needs to handle milliseconds and truncated timestamps?
+            let timestamp = NaiveDateTime::parse_from_str(raw_timestamp, "%Y%m%d%H%M%S")?.and_utc();
+
+            (timestamp, vals)
+        };
+
+        for (i, val) in vals.enumerate() {
+            // TODO: should we do some smart bounds-checking??
+            let col = columns[i].clone();
+
+            // rejection is acceptable here, because things we don't catch should
+            // be covered by the checked queue
+            let param_entry = reference_params.get(&col.param_code);
+
+            let (sensor, level) = col.sensor_and_level.unwrap_or((0, 0));
+
+            let val = val.trim();
+
+            // // TODO: is this right?
+            // // Ignore empty values
+            if val.is_empty() {
+                continue;
+            }
```

Comment:
In the obsinn fixes branch I have something like [this](https://github.com/metno/lard/blob/9d92a6e5064db68c5ee01ff1707ec16344360d3f/ingestion/src/kldata.rs#L164-L171), maybe `Obstype` variants should be options?

Reply:
See noted intention to discuss with Søren. If these indicate deletion, we should actually do a deletion

Reply:
I saw the note, and recall the discussion... I think they really are deletions, but obviously we need to clarify with søren.

Reply:
Ah sorry!